### DISCUSSION TITLE
Verifying if interaction user matches expected target

### DISCUSSION THREAD

**Discussion by Legit4K** [discordjs/discord.js#7525] [Q&A] [answered]
**Context:** I have an interactive discord game bot, the **/start** command is used to display a "pre-game" message which is actually just a blank embed board with 2 buttons, namely **Play** and **Tutorial**. The rest of the game is handled in the `interactionCreate` event listener. When the **Play** button is pressed, the orignal message with that embed is updated and a new button is displayed -> **Give Up**.

**Problem:** I'm not sure how I could test if the interaction user matches the same person who ran the initial start command which is handled in its own command file. Keep in mind, this **Give Up** Button is handled within the `interactionCreate` event listener.

Im tired and its like 3 in the morning but I'll rest better after knowing there's a viable solution 😅

**Comment by imranbarbhuiya** [comment, marked as answer]
U can either use a collector with filter. Or for some reason, if u want to listen to `interactionCreate` then u can add the user id in the button customId. And then check with that customId. Or u can also do `interaction.message.interaction.user` to get the user who executed `start` command if it's a slash command.

**Reply by Legit4K** [reply]
Hmm, I'll try the last one

**Reply by Legit4K** [reply]
The last option worked wonders for me, thanks a ton 👍
